File: C#/13.cs
public class Solution {
    public int RomanToInt(string s) {
        int result = 0;
        for (int i = 0; i < s.Length; i++) {
            if (i < s.Length - 1 && GetValue(s[i]) < GetValue(s[i+1])) {
                result -= GetValue(s[i]);
            } else {
                result += GetValue(s[i]);
            }
        }
        return result;
    }
    
    private int GetValue(char c) {
        switch (c) {
            case 'I': return 1;
            case 'V': return 5;
            case 'X': return 10;
            case 'L': return 50;
            case 'C': return 100;
            case 'D': return 500;
            case 'M': return 1000;
            default: return 0;
        }
    }
}

File: python/13.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_dict = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        result = 0
        prev_value = 0
        for i in range(len(s)-1, -1, -1):
            value = roman_dict[s[i]]
            if value < prev_value:
                result -= value
            else:
                result += value
            prev_value = value
        return result